src/messages/mod.rs

```diff
@@ -231,14 +276,51 @@ impl Debug for SecurityMetadata {
 }
 
 /// Wrapper around a routing message, signed by the originator of the message.
-#[derive(Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
+#[derive(Debug, Eq, PartialEq, Clone, Hash)]
+pub struct PartialSignedRoutingMessage {
+    /// Destination location
+    pub dst: Location,
+}
+
+impl<'de> Deserialize<'de> for PartialSignedRoutingMessage {
+    fn deserialize<D: Deserializer<'de>>(deserialiser: D) -> std::result::Result<Self, D::Error> {
+        let dst: Location = Deserialize::deserialize(deserialiser)?;
+        Ok(PartialSignedRoutingMessage { dst })
+    }
+}
```

Comment:
Is this manual impl necessary? Why not just `derive(Deserialize)` instead?

Reply:
I think it is, the struct derive is different, but I'll try.

Reply:
It does work, though it feels really wrong, we serialize the SigneRoutingMessage as a tuple it seem more reasonable to deserialize it as a tuple. Cannot have tuple of 1, so next best thing I though was as the sole member.

Reply:
It works with bincode (and possibly some other formats) because it doesn't store any tags, headers, etc.. So a struct of one field is serialized the same as that field by itself. It doesn't work in general though. We are unlikely to switch to other serialization format any time soon so this should be fine. Best to have it covered by tests though, which you do so we should be good.

Reply:
do you mean I should still just use derive(Deserialize)?

Reply:
Also, a tuple of N elements is serialized as just the first element, then the second, etc... There is no header saying how many elements there are and there are no separators between the elements. This is because those information can be inferred from the type you are deserializing into.

Reply:
> do you mean I should still just use derive(Deserialize)?

Yes. It achieves the same effect with less (hand-written) code.